tests/distributed/test_partition.py

```diff
@@ -120,24 +120,50 @@ def verify_hetero_graph(g, parts):
         assert len(uniq_ids) == g.number_of_edges(etype)
     # TODO(zhengda) this doesn't check 'part_id'
 
-def verify_graph_feats(g, part, node_feats):
+def verify_graph_feats(g, gpb, part, node_feats, edge_feats):
     for ntype in g.ntypes:
         ntype_id = g.get_ntype_id(ntype)
+        inner_node_mask = _get_inner_node_mask(part, ntype_id)
+        inner_nids = F.boolean_mask(part.ndata[dgl.NID],inner_node_mask)
+        ntype_ids, inner_type_nids = gpb.map_to_per_ntype(inner_nids)
+        partid = gpb.nid2partid(inner_type_nids, ntype)
+        assert np.all(F.asnumpy(ntype_ids) == ntype_id)
+        assert np.all(F.asnumpy(partid) == gpb.partid)
+
+        orig_id = F.boolean_mask(part.ndata['orig_id'], inner_node_mask)
+        local_nids = gpb.nid2localnid(inner_type_nids, gpb.partid, ntype)
+
         for name in g.nodes[ntype].data:
             if name in [dgl.NID, 'inner_node']:
                 continue
-            inner_node_mask = _get_inner_node_mask(part, ntype_id)
-            inner_nids = F.boolean_mask(part.ndata[dgl.NID],inner_node_mask)
-            min_nids = F.min(inner_nids, 0)
-            orig_id = F.boolean_mask(part.ndata['orig_id'], inner_node_mask)
             true_feats = F.gather_row(g.nodes[ntype].data[name], orig_id)
-            ndata = F.gather_row(node_feats[ntype + '/' + name], inner_nids - min_nids)
+            ndata = F.gather_row(node_feats[ntype + '/' + name], local_nids)
             assert np.all(F.asnumpy(ndata == true_feats))
 
+    for etype in g.etypes:
+        etype_id = g.get_etype_id(etype)
+        inner_edge_mask = _get_inner_edge_mask(part, etype_id)
+        inner_eids = F.boolean_mask(part.edata[dgl.EID],inner_edge_mask)
+        etype_ids, inner_type_eids = gpb.map_to_per_etype(inner_eids)
+        partid = gpb.eid2partid(inner_type_eids, etype)
+        assert np.all(F.asnumpy(etype_ids) == etype_id)
+        assert np.all(F.asnumpy(partid) == gpb.partid)
+
+        orig_id = F.boolean_mask(part.edata['orig_id'], inner_edge_mask)
+        local_eids = gpb.eid2localeid(inner_type_eids, gpb.partid, etype)
+
+        for name in g.edges[etype].data:
+            if name in [dgl.EID, 'inner_edge']:
```

Comment:
Shall we put inner_edge as dgl.INNER_EDGE?

Reply:
i'm not sure.